llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4183,17 +4193,32 @@ LoopVectorizationCostModel::getMaxLegalScalableVF(unsigned MaxSafeElements) {
     reportVectorizationInfo("Scalable vectorization is not supported "
                             "for all element types found in this loop.",
                             "ScalableVFUnfeasible", ORE, TheLoop);
-    return ElementCount::getScalable(0);
+    return false;
   }
 
+  IsScalableVectorizationAllowed = true;
+  return true;
+}
+
+ElementCount
+LoopVectorizationCostModel::getMaxLegalScalableVF(unsigned MaxSafeElements) {
+  if (!isScalableVectorizationAllowed())
+    return ElementCount::getScalable(0);
+
+  auto MaxScalableVF = ElementCount::getScalable(
+      std::numeric_limits<ElementCount::ScalarTy>::max());
```

Comment:
This is fine, preserving current behavior. Would be good if some assert could be added, independently, to make sure that `MaxSafeElements` corresponds to `MaxScalableVF` if isSafeForAnyVectorWidth(), as the former is ignored in this case, which returns the latter.

Reply:
As I said before, this requires looking into the internals of LAA. No idea how safely do it here

Reply:
Actually, probably we can just use `auto MaxScalableVF = ElementCount::getScalable(
      MaxSafeElements);` here